nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-processor-configuration.js

```diff
@@ -741,8 +742,8 @@
                         }
                     });
 
-                    // show the execution node option if we're cluster or we're currently configured to run on the primary node only
-                    if (nfClusterSummary.isClustered() || executionNode === 'PRIMARY') {
+                    // show the execution node option if we're clustered and execution node is not restricted to run only in primary node
+                    if (nfClusterSummary.isClustered() && executionNodeRestricted !== true) {
```

Comment:
I think we still need the executionNode === 'PRIMARY' here:
```
if ((nfClusterSummary.isClustered() && executionNodeRestricted !== true) || executionNode === 'PRIMARY') {
```
This way, if running in standalone mode, but the processor is marked with an ExecutionNode of Primary Node (which may be the case if instantiating a template from a cluster, or if if copying a flow.xml.gz over or something like that) we still have the ability to change it to 'All Nodes'.

Reply:
No. If we add `executionNode === 'PRIMARY'` then the UI will show the dropdown menu for `ExecutionNodes`. Moreover, why should we need the ability to change to `All Nodes` in the said scenario because it is running in standalone mode, right? Primary Node or All Nodes doesn't make sense in standalone mode. Correct?

Reply:
We can make it this: `if ((nfClusterSummary.isClustered() && executionNodeRestricted !== true) || (!nfClusterSummary.isClustered() && executionNode === 'PRIMARY'))` but I don't understand why we are doing the executionNode === 'PRIMARY' check, for the same reason I had mentioned above.

Reply:
I believe the `executionNode === 'PRIMARY'` was in place to ensure the currently configured value is shown. If the current value is set to PRIMARY, but this instance is no longer clustered we need to render that fact. Once the user reconfigures this value, they will no longer be able to select this option (since the node isn't clustered and executionNode would be ALL). Hope this makes sense.

Reply:
Understood. So I think `if ((nfClusterSummary.isClustered() && executionNodeRestricted !== true) || (!nfClusterSummary.isClustered() && executionNode === 'PRIMARY'))` will do the job. Correct?

Reply:
@mcgilman With the current state of the commit, if a processor previously didn't have `executionNodeRestricted` and the `executionNode` was `ALL`, it will still be hidden. Now I understand that shouldn't be the case. Thanks for the clarification so I think the following will address this:

    var getExecutionNodeOptions = function (processor) {
        return [{
            text: 'All nodes',
            value: 'ALL',
            description: 'Processor will be scheduled to run on all nodes',
            disabled: processor.executionNodeRestricted === true
        }, {
            text: 'Primary node',
            value: 'PRIMARY',
            description: 'Processor will be scheduled to run only on the primary node',
            disabled: !nfClusterSummary.isClustered() && processor.config['executionNode'] === 'PRIMARY'
        }];
    };

And changing the code to show when `execution-node-options` gets shown in the UI to the following:

    if ((nfClusterSummary.isClustered() && executionNodeRestricted !== true) ||
        (!nfClusterSummary.isClustered() && executionNode === 'PRIMARY') ||
        (executionNodeRestricted === true && executionNode === 'ALL')) {
        $('#execution-node-options').show();
    } else {
        $('#execution-node-options').hide();
    }

There are two much checks in the `if` but I think they are needed to address this. If that can be optimized or modified, do let me know.

Reply:
@markap14 @mcgilman I've tested the above and verified it to be working as expected. Request you to review this approach and if you don't have any concerns, I'll make the changes and commit it.

Reply:
@mcgilman @markap14 Bump... Any update? If you're okay, I can make the commit.

Reply:
@zenfenan Sorry, I missed the mentions here. Always feel free to push subsequent commits as we're working through the PR process. It's much easier to review when I pull down the changes locally. What your suggesting above is exactly what I was thinking. I agree that the conditional is a bit confusing. I wonder if we attempt to break it up a little bit if it would be more clear. Maybe something like

`
                    if (nfClusterSummary.isClustered()) {
                        if (executionNodeRestricted !== true || executionNode === 'ALL') {
                            $('#execution-node-options').show();
                        } else {
                            $('#execution-node-options').hide();
                        }
                    } else {
                        if (executionNode === 'PRIMARY') {
                            $('#execution-node-options').show();
                        } else {
                            $('#execution-node-options').hide();
                        }
                    }
`

Thoughts?

Reply:
Yep. It's better. I've added it and pushed the commit. Appreciate if you could test it out and confirm the changes :)